### PR TITLE
PT-6909 Extracted spec loading to config and added some tests

### PR DESCRIPTION
**Pull request by vgarvardt**
First piece extracted from bigger brother https://github.com/hellofresh/klepto/pull/110

In this PR:
- improved error reporting
- moved config loading to config package and covered it with basic tests
- fixed `init` command as generated toml file has invalid structure - go sruct embedding for some reasons works differently for writing toml and reading it

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -54,12 +65,45 @@ type (
...
+		// Use config file from the flag.
+		log.Debugf("Reading config from %s ...", configPath)
+		viper.SetConfigFile(configPath)
+	} else {
+		log.Debugf("Reading config from %s ...", DefaultConfigFileName)
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, wErrors.Wrap(err, "can't find current working directory")
+		}
```
Comment (boekkooi-fresh) [inline pkg/config/config.go:90]: Should we put this in a `func init()` and set `DefaultConfigFileName` correctly from the get go?

File: pkg/config/config.go [hunk 2]
```diff
@@ -54,12 +65,45 @@ type (
...
+
+		viper.SetConfigName(".klepto")
+		viper.AddConfigPath(cwd)
+		viper.AddConfigPath(".")
```
Comment (boekkooi-fresh) [inline pkg/config/config.go:94]: Why not use `viper.SetConfigFile` here as well?

File: pkg/dumper/dumper.go [hunk 3]
```diff
@@ -55,10 +57,10 @@ func NewDumper(opts ConnOpts, rdr reader.Reader) (dumper Dumper, err error) {
 	})
 
 	if dumper == nil && err == nil {
-		err = errors.New("no supported driver found")
+		err = fmt.Errorf("no supported driver found for dumper DSN %q", opts.DSN)
 	}
 
-	err = errors.Wrapf(err, "could not create dumper for dsn: '%v'", opts.DSN)
+	err = wErrors.Wrapf(err, "could not create dumper for DSN: %q", opts.DSN)
```
Comment (boekkooi-fresh) [inline pkg/dumper/dumper.go:63]: Seems a little double up with the error in line 60 .... Maybe put it in an `else` or remove the DSN from line 60 error

Comment (vgarvardt) [reply pkg/dumper/dumper.go:63]: On line 60 it check of the driver was found (main part is `dumper == nil`) and here it wraps the error if any - the trick is that if the error is `nil` wrapped result would also be `nil` - it is not very obvious, I had to dig into the code to understand it. It worth rewriting driver search part, but here I just slightly improved error reporting as I could not understad anything from current when started to reproduce the problem and debug it.

File: pkg/config/config.go (hunk 2 above)
Comment (vgarvardt) [reply pkg/config/config.go:94]: I'll remove this part in the next PRs completely as with default value set in flags there is no way to get here. But now I just left the function as I got in in another file.

File: pkg/config/config.go (hunk 1 above)
Comment (vgarvardt) [reply pkg/config/config.go:90]: This part will go away completely. Actually let me just remove it here.

Comment (vgarvardt) [reply pkg/config/config.go]: removed

File: pkg/config/config.go (hunk 2 above)
Comment (vgarvardt) [reply pkg/config/config.go]: removed

File: pkg/dumper/dumper.go [hunk 4]
```diff
@@ -55,10 +57,10 @@ func NewDumper(opts ConnOpts, rdr reader.Reader) (dumper Dumper, err error) {
 	})
 
 	if dumper == nil && err == nil {
-		err = errors.New("no supported driver found")
+		err = fmt.Errorf("no supported driver found for dumper DSN %q", opts.DSN)
```
Comment (boekkooi-fresh) [inline pkg/dumper/dumper.go:60]:
```suggestion
		err = fmt.Errorf("no supported driver found for dumper")
```
Based on https://github.com/hellofresh/klepto/pull/111/files#r402352477

File: pkg/dumper/dumper.go (hunk 3 above)
Comment (boekkooi-fresh) [reply pkg/dumper/dumper.go:63]: Added a suggestion for line 60 :smile:

**Comment by boekkooi-fresh** [comment]
:+1: LGTM

**Comment by samacoff** [comment]
+1

**Pull request merged**